core/inc/message_queue.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef MESSAGE_QUEUE_H
+#define MESSAGE_QUEUE_H
+
+#include "message.h"
+
+#include "azure_c_shared_utility/crt_abstractions.h"
+#include "azure_c_shared_utility/umock_c_prod.h"
+
+#ifdef __cplusplus
+#include <cstddef>
+#include <cstdbool>
+extern "C"
+{
+#else
+#include <stddef.h>
+#include <stdbool.h>
+#endif
+
+typedef struct MESSAGE_QUEUE_TAG* MESSAGE_QUEUE_HANDLE;
+
+/* creation */
+MOCKABLE_FUNCTION(, MESSAGE_QUEUE_HANDLE, MESSAGE_QUEUE_create);
```

Comment:
This isn't exported out of the gateway library, does it need doxygen documentation?